src/features/posts/postSlice.js

```diff
@@ -0,0 +1,80 @@
+import { sub } from 'date-fns'
+import { createSlice, nanoid, createAsyncThunk } from '@reduxjs/toolkit'
+import { client } from '../../api/client'
+
+const initialState = {
+  posts: [],
+  status: 'idle',
+  error: null
+}
+export const fetchPosts = createAsyncThunk('posts/fetchPosts', async () => {
+  const response = await client.get('/fakeApi/posts')
+  return response.data
+})
+
+export const addNewPost = createAsyncThunk(
+  'posts/addNewPost',
+  // The payload creator receives the partial `{title, content, user}` object
+  async initialPost => {
+    // We send the initial data to the fake API server
+    const response = await client.post('/fakeApi/posts', initialPost)
+    // The response includes the complete post object, including unique ID
+    return response.data
+  }
+)
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
```

Comment:
Found the issue. It was tricky but I found that nothing was reaching our extra reducers so there had to be something wrong with the setup. The `reducers` and `extraReducers` objects need to be separated. Currently `extraReducers` is inside of `reducers`. Here's a snippet of the fixed code where `reducers` and `extraReducers` are separated.
```
const postsSlice = createSlice({
  name: 'posts',
  initialState,
  reducers: {
    postAdded: {
      reducer(state, action) {
        state.posts.push(action.payload)
      },
      prepare(title, content, userId) {
        return {
          payload: {
            id: nanoid(),
            date: new Date().toISOString(),
            title,
            content,
            user: userId,
          },
        }
      },
    },
  },
  extraReducers(builder) {
    console.log('builder eh')
    builder
      .addCase(fetchPosts.pending, (state, action) => {
        console.log('pending state', state)
        state.status = 'loading'
      })
      .addCase(fetchPosts.fulfilled, (state, action) => {
        console.log('state', state)
        state.status = 'succeeded'
        state.posts = state.posts.concat(action.payload)
      })
      .addCase(fetchPosts.rejected, (state, action) => {
        state.status = 'failed'
        state.error = action.error.message
      })
      .addCase(addNewPost.fulfilled, (state, action) => {
        state.posts.push(action.payload)
      })
  },
  postUpdated(state, action) {
    const { id, title, content } = action.payload
    const existingPost = state.posts.find((post) => post.id === id)
    if (existingPost) {
      existingPost.title = title
      existingPost.content = content
    }
  },
  reactionAdded(state, action) {
    const { postId, reaction } = action.payload
    const existingPost = state.posts.find((post) => post.id === postId)
    if (existingPost) {
      existingPost.reactions[reaction]++
    }
  },
})

```